datahub/interaction/email_processors/processors.py

```
@@ -151,9 +195,9 @@ def process_email(self, message):
         email_parser = CalendarInteractionEmailParser(message)
         try:
             interaction_data = email_parser.extract_interaction_data_from_email()
-        except ValidationError as exc:
-            self._notify_meeting_ingest_failure(message, exc.messages)
-            return (False, exc.message)  # noqa: B306
+        except InvalidInviteError as exc:
```

Note: I decided it would be better to wholly depend on a helper method than using separate except statements as this was the most DRY approach.

---

Perhaps following is a better way handling exceptions instead of `_handle_invalid_invite`:

```
def _extract_interaction_from_email(email_paresr):
    try:
        interaction_data = email_parser.extract_interaction_data_from_email()
    except InvalidInviteError as exc:
        # handle the exception..
        pass
```

Essentially encapsulate the function that is raising the exception in one that catches and handles it.

---

I agree with that approach, but it gets a bit complicated when we need to transform the exception in to an early return value in `process_email()`.

I think this loops back to your earlier comment in a previous PR that we should change the contract for email processors to raise exceptions for exceptional cases rather than returning a pair tuple of `(succeeded, message)`.  I'd still like to dedicate some time to this but my feeling is that it's out of scope for this PR.